### ISSUE TITLE
[FR] Return Orders Improvements

### ISSUE THREAD

**Issue by timmyhadwen** [inventree/InvenTree#8065]
### Please verify that this feature request has NOT been suggested before.

- [X] I checked and didn't find a similar feature request

### Problem statement

We use return orders to manage repairs for our products from time to time. Currently the process is as follows:
- Create return order
- Receive line items (this marks them as Quarantined)
- Repair items
- Remove stock of any parts used (manually)
- Mark each line item as OK
- Assign each line item to Customer

This process is long and prone to human error

### Suggested solution

It would be great if the status in the return order for each line item did these steps. Examples

A is returned, repaired and sent back to customer
- Automatically mark as OK, and reassign back to customer

B is returned, unable to be repaired and is replaced.
- Automatically mark as LOST (or other status) and assign a replacement stock item to the customer.

C is returned, and replaced without attempting repair (important customer needs working hardware immediately)
- Automatically mark as "needs repair", and assign a replacement stock item to the customer.

### Describe alternatives you've considered

we will continue to use the current method

### Examples of other systems

N/A

### Do you want to develop this?

- [ ] I want to develop this.

**Comment by SchrodingersGat** [comment]
@timmyhadwen agreed that the return order functionality could be improved, but I must admit I don't quite follow what you are suggesting here. Can you expand on / explain it a bit further?

**Comment by timmyhadwen** [comment]
Yeah no problem happy to discuss more but here is what I think would make the most sense.

When items in a return are received, they should (at a later stage when they are processed/tested etc) be able to be given status which corresponds to actions. For example:

REPAIRED or REJECTED - Should leave a with repair details (either was repaired and x was replaced, or no issue was not repaired), and then assign this back to the customer in a shipment allocated to the return order.

REPLACED - Should mark the returned item as Quarantined, and then allow us to allocate a new item to the customer (ideally without having to create seperate sales order etc). Have this item allocated to a shipment linked to the return order.

Im sure there are other cases that would have different actions but these are the ones that we deal with all the time with our products.

I think it would also be useful to be able to have a status for TEST REQUIRED etc as part of the receiving process, but im not sure how this would work tbh. Is probably a separate issue.

**Comment by timmyhadwen** [comment]
@SchrodingersGat Was having a look at how FishBowl handles RMAs. Might be worth a read to aid discussions:
https://help.fishbowlinventory.com/s/article/RMA

Extract which explains basically what Im asking for :)
**4. Resolve an RMA**

Once a returned item has been received, the RMA item can be marked as resolved by highlighting the line item in the Icon.RMA.png RMA module and selecting the appropriate resolution from the Resolution field. After resolving the item, click the Save button on the module toolbar. The [Resolution tab of the](https://help.fishbowlinventory.com/s/article/RMA#Module_Options_Resolution_tab) Icon.RMA.png [RMA module options](https://help.fishbowlinventory.com/s/article/RMA#Module_Options_Resolution_tab) also provides the ability to automatically resolve items as soon as they are received.

The Resolution options can be customized by opening the Icon.Custom Field.png [Custom Field module](https://help.fishbowlinventory.com/s/article/Custom-Field), clicking the Custom Lists Button.png Lists button, and then selecting RMA Resolution. The RMA Issue list can be modified in the same way. Items that are added to custom 

**Ship a replacement product** - If the replacement product was not [cross shipped](https://help.fishbowlinventory.com/s/article/RMA#Cross_ship_products), click the Send Button.png Send button in the Icon.RMA.png RMA module to create a new sales order with the resolved items. The new sales order will appear on the Links tab and can now be picked, packed, and shipped to the customer.

**Refund a customer payment** - When a replacement product is not shipped to the customer, the original customer payment can be refunded by loading the credit return sales order in the Icon.Sales Order.png [Sales Order module](https://help.fishbowlinventory.com/s/article/Sales-Order) and then clicking the [Payment button in the Sales Order module](https://help.fishbowlinventory.com/s/article/Sales-Order#Take_a_payment).
Once all items have been resolved, the RMA will have a Fulfilled status.

**Comment by SchrodingersGat** [comment]
@timmyhadwen I see! Would you be willing to implement this / contribute towards implementation?

**Comment by timmyhadwen** [comment]
Yeah happy to contribute. Planning on jumping on a few of the starter tickets to get comfy and then come back to this one perhaps.

**Comment by SchrodingersGat** [comment]
@timmyhadwen awesome 👍

**Comment by SchrodingersGat** [comment]
Still important

**Comment by SchrodingersGat** [comment]
Hey @timmyhadwen it's been a while - are you still interested in developing this?
